[FEAT] 워크플로우 단계별 분리

Baguette-bbang opened the pull request:
## :hash: 연관된 이슈
#440 

## :memo: 작업 내용
- 단일 배포 스크립트를 기능별로 모듈화
- 각 단계별로 독립적인 스크립트로 분리하여 디버깅과 개별 테스트를 용이하도록 변경했습니다.
```
scripts/                                    # 배포 스크립트 모듈화
├── setup-system.sh                         # 시스템 환경 자동 설정
├── setup-ssl-deployment.sh                 # SSL 배포 관리
├── deploy-application.sh                   # 애플리케이션 배포 + 이미지 정리
├── setup-ssl-renewal.sh                    # SSL 자동 갱신 설정
└── verify-deployment.sh                    # 배포 상태 종합 검증
```

추가적인 내용들은 아래 문서를 확인해주세요.
https://www.notion.so/CI-CD-SSL-200246cdb8768063b058d8aaa7521156?pvs=4

Baguette-bbang committed:
feat : ci/cd 스크립트 용도에 따른 분리

1117mg reviewed (commented):
안정적인 운영에 힘써주셔서 감사합니다 고생하셨습니다!

miu1217 reviewed (approved):
수고하셨습니다 항상 서버는 영민님 덕분에 걱정 없이 작업하고 있습니다

ddhi7 reviewed (approved):
고생하셨습니다!

Baguette-bbang committed:
feat : 확장 가능한 헬스체크 시스템으로 배포 로직 개선

- 기존 하드코딩된 헬스체크를 설정 기반 시스템으로 분리
- health-check.sh: 설정 파일 기반으로 서비스별 헬스체크 수행
- 새 서비스 추가 시 설정 파일만 수정하면 자동 헬스체크 포함
- HTTP, Redis, 컨테이너, 커스텀 명령어 등 다양한 체크 타입 지원
- deploy-application.sh에서 분리된 헬스체크 스크립트 활용
- 자동 설정 파일 생성으로 첫 실행 시 즉시 사용 가능
